### ISSUE TITLE
Review JaxIntro tutorial 

### ISSUE THREAD

**Issue by rameshnatarajanus** [YCCS-Summer-2023-DDNMA/project#33]
[JaxIntro ](https://github.com/YCCS-Summer-2023-DDNMA/project/blob/main/ramesh_natarajan/JaxIntro.ipynb)

As before, you can play around with the notebook and make/save copies in your own github folders.

Jax can be thought of as a way of writing Python programs with a lot of numpy computations, with the following key differences.

Jax programs implement algorithms using pure python functions (no function state or side effects), and numpy which is implemented through the jax library jax.numpy  has some gotchas like non-mutable arrays and type casting restrictions which you should know). 

The key features of Jax you should be familiar with are:

- Built-in transformations to support ML  functionality, performance and scaling
- gradients and higher-order derivatives (grad, jacfwd, jacrev, hessian)
- code transformation via  XLA  optimized for GPU/TPU  (jit)
- single-example to batch execution (vmap)
- multi-device data-parallel execution on TPU’s (pmap)
- differential privacy with gradient clipping + noise (vmap, dp-sgd)
- flexible data structures for model parameters (pytrees)
